services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/data/HaasController.java

```diff
@@ -0,0 +1,67 @@
+package us.dot.its.jpo.ode.api.controllers.data;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.http.MediaType;
+
+import us.dot.its.jpo.ode.api.accessors.haas.HaasLocationDataRepository;
+import us.dot.its.jpo.ode.api.models.PaginatedGeoJsonResponse;
+import us.dot.its.jpo.ode.api.models.haas.HaasLocation;
+
+@Slf4j
+@RestController
+@ConditionalOnProperty(value = { "enable.api", "enable.haas" }, havingValue = "true", matchIfMissing = false)
+@ApiResponses(value = {
+                @ApiResponse(responseCode = "401", description = "Unauthorized"),
+                @ApiResponse(responseCode = "500", description = "Internal Server Error")
+})
+public class HaasController {
+
+        private final HaasLocationDataRepository haasLocationDataRepository;
+
+        @Autowired
+        public HaasController(HaasLocationDataRepository haasLocationDataRepository) {
+                this.haasLocationDataRepository = haasLocationDataRepository;
+        }
+
+        @Operation(summary = "HAAS Alert Locations", description = "Returns HAAS alert locations in GeoJSON format with pagination")
+        @RequestMapping(value = "/data/haas/locations", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
suggestion(non-blocking): use the specific `@GetMapping` for clarity of intent
```suggestion
        @GetMapping(value = "/locations", produces = MediaType.APPLICATION_JSON_VALUE)
```